server/src/main/java/org/opensearch/index/translog/TranslogManager.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+/**
+ * The interface that orchestrates Translog operations and manages the {@link Translog} and interfaces with the Engine
+ *
+ * @opensearch.internal
+ */
+public interface TranslogManager {
```

Comment:
Are there any methods that would be common across different implementations, like `getTranslog`? In that case, would it help if `TranslogManager` is abstract class instead of an Interface?

Reply:
The two implementations `WriteOnlyTranslogManager` extends `InternalTranslogManager` while the `NoOpTranlogManager` doesn't need one so there is a re-use already(limited) where it's needed